vignettes/piggyback.Rmd

```diff
@@ -36,113 +33,202 @@ number of files or bandwidth to deliver them.
-  |======================================================| 100%
-```
-```r
+```{r}
+pb_download(
+  file = "iris2.tsv.gz", 
+  dest = tempdir(),
+  repo = "cboettig/piggyback-tests",
+  tag = "v0.0.1"
+  )
+#> ℹ Downloading "iris2.tsv.gz"...
+#>   |======================================================| 100%
 fs::dir_tree(tempdir())
+#> /tmp/RtmpWxJSZj
+#> └── iris2.tsv.gz
 ```
-```
-/tmp/RtmpWxJSZj
-└── iris2.tsv.gz
-```
-
-**Tips:**
-
-1. Whenever you are working from a location inside a git repository corresponding 
-to your GitHub repo, you can simply omit the `repo` argument and it will be detected 
-automatically. 
-2. Likewise, if you omit the release `tag`, `pb_download` will simply pull data 
-from most recent release (`latest`).  
-3. You can omit `tempdir()` if you are using an RStudio Project (`.Rproj` file) 
-in your repository: download locations will be relative to Project root.  
-`tempdir()` is used throughout the examples only to meet CRAN policies and is 
-unlikely to be the choice you actually want here.  
-
-4. Omit the file name to download all assets connected with a given release.  
-```r
-pb_download(repo = "cboettig/piggyback-tests",
-            tag = "v0.0.1",
-            dest = tempdir())
-```
-```
-ℹ Downloading "diamonds.tsv.gz"...
-  |======================================================| 100%
-ℹ Downloading "iris.tsv.gz"...
-  |======================================================| 100%
-ℹ Downloading "iris.tsv.xz"...
-  |======================================================| 100%
-```
-```r
+Some default behaviors to know about:
+
+1. The `repo` argument in most piggyback functions will default to detecting the 
+   relevant GitHub repo based on your current working directory's git configs, 
+   so in many cases you can omit the `repo` argument.
+2. The `tag` argument in most functions defaults to "latest", which typically 
+   refers to the most recently created release of the repository, unless there 
+   is a release specifically named "latest" or if you have marked a different
+   release as "latest" via the GitHub UI. 
+3. The `dest` argument defaults to your current working directory (`"."`). We
+   use `tempdir()` to meet CRAN policies for the purposes of examples.
+4. The `file` argument in `pb_download` defaults to NULL, which will download
+   all files connected to a given release:
+```{r}
+pb_download(
+  repo = "cboettig/piggyback-tests",
+  tag = "v0.0.1",
+  dest = tempdir()
+)
+#> ℹ Downloading "diamonds.tsv.gz"...
+#>   |======================================================| 100%
+#> ℹ Downloading "iris.tsv.gz"...
+#>   |======================================================| 100%
+#> ℹ Downloading "iris.tsv.xz"...
+#>   |======================================================| 100%
 fs::dir_tree(tempdir())
+#> /tmp/RtmpWxJSZj
+#> ├── diamonds.tsv.gz
+#> ├── iris.tsv.gz
+#> ├── iris.tsv.xz
+#> └── iris2.tsv.gz
 ```
+5. The `use_timestamps` argument defaults to TRUE - notice that above, 
```

Comment:
noticed while explaining this that use_timestamps and overwrite are not merged as they are in pb_upload, raised new issue in #119